openedx/features/edly/middleware.py

```diff
@@ -0,0 +1,24 @@
+"""
+Edly Organization Access Middleware.
+"""
+from logging import getLogger
+from django.http import Http404
+
+from openedx.features.edly.utils import user_has_edly_organization_access
+
+logger = getLogger(__name__)
+
+
+class EdlyOrganizationAccessMiddleware(object):
+    """
+    Django middleware to validate edly user organization access based on request.
+    """
+
+    def process_request(self, request):
+        """
+        Validate logged in user's access based on request site and its linked edly sub organization.
+        """
+
+        if request.user.is_authenticated and not user_has_edly_organization_access(request):
```

Comment:
First we need to check if the `request.user` is superuser then we need to allow the user to proceed with the request. Something like (please verify for anonymous users):

```python
user_is_authenticated = request.user.is_authenticated
user_is_superuser = request.user.is_superuser)
if user_is_authenticated and not user_is_superuser and not user_has_edly_organization_access(request)):
```

So that the MAU's report request which is requested on behalf of superuser don't get affect.

Please add related test as well.

Reply:
Good catch!

Reply:
Condition for `Super User` in Middleware and relevant Unit Test added.